server/src/main/java/org/elasticsearch/index/mapper/ArrayTypeParser.java

```diff
@@ -42,8 +42,6 @@ public Mapper.Builder parse(String name, Map<String, Object> node, ParserContext
         String typeName = (String) innerNode.get("type");
         if (typeName == null && innerNode.containsKey("properties")) {
             typeName = ObjectMapper.CONTENT_TYPE;
-        } else if (ArrayMapper.CONTENT_TYPE.equalsIgnoreCase(typeName)) {
-            throw new MapperParsingException("nested arrays are not supported");
         }
```

Comment:
There is also `BuilderFactory.detectInnerMapper` which can throw the same error - I guess can be removed as well.

Reply:
I checked the only usage, it's basically the only implementation of `DynamicArrayFieldMapperBuilderFactory`.

It's used in `DocumentParser` which afaik is used for reading from the translog. 

Just loudly thinking: looks like to hit this code, we need a dynamic mapping update inserting a multi-dimensional array value and imitate that's it's written to translog? I wonder if we can setup a test case in `NestedArrayTypeTest`? 

In case scenario I'm describing is invalid, is it possible to find a valid scenario and hit this code path?

Reply:
For the record: as discussed, we can do it as a follow up. For now creation of nested array column (or sub-column) via dynamic mapping updated is disallowed.